File: filter_products.py
import json

file_path = r"c:\xampp1\htdocs\athens\products.js"

with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

prefix = "const products = "
suffix = ";"

content_trimmed = content.strip()
if content_trimmed.endswith(suffix):
    content_trimmed = content_trimmed[:-1]

try:
    json_str = content_trimmed[content_trimmed.find(prefix) + len(prefix):]
    products = json.loads(json_str)
except Exception as e:
    print(f"Error parsing JSON: {e}")
    exit(1)

category_counts = {}
filtered_products = []

for p in products:
    cat = p.get("category", "Unknown")
    if cat not in category_counts:
        category_counts[cat] = 0
    
    if category_counts[cat] < 15:
        filtered_products.append(p)
        category_counts[cat] += 1

js_content = prefix + json.dumps(filtered_products, indent=4) + suffix + "\n"

with open(file_path, "w", encoding="utf-8") as f:
    f.write(js_content)

print(f"Successfully reduced products from {len(products)} to {len(filtered_products)}.")

File: filter_products.php
<?php
$file_path = "products.js";
$content = file_get_contents($file_path);

$prefix = "const products = ";
$suffix = ";";

$content_trimmed = trim($content);
if (substr($content_trimmed, -1) === ';') {
    $content_trimmed = substr($content_trimmed, 0, -1);
}

$json_str = substr($content_trimmed, strpos($content_trimmed, $prefix) + strlen($prefix));
$products = json_decode($json_str, true);

if ($products === null) {
    die("Error parsing JSON\n");
}

$category_counts = [];
$filtered_products = [];

foreach ($products as $p) {
    $cat = isset($p['category']) ? $p['category'] : "Unknown";
    if (!isset($category_counts[$cat])) {
        $category_counts[$cat] = 0;
    }
    
    if ($category_counts[$cat] < 15) {
        $filtered_products[] = $p;
        $category_counts[$cat]++;
    }
}

$js_content = $prefix . json_encode($filtered_products, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES) . $suffix . "\n";
file_put_contents($file_path, $js_content);

echo "Successfully reduced products from " . count($products) . " to " . count($filtered_products) . ".\n";
?>
